pkg/tbtc/chain.go

```diff
@@ -125,6 +152,7 @@ type Chain interface {
 	GroupSelectionChain
 	DistributedKeyGenerationChain
 	BridgeChain
+	RelayChain
```

Comment:
I would consider organizing it as:
```
type MaintainerChain interface {
  BitcoinDifficultyMaintainer() (chain.BitcoinDifficultyMaintainer, error)
}

type BitcoinDifficultyMaintainer interface {
  Ready (bool, error)
  AuthorizationRequired() (bool, error)
  // (...)
}
```

This way, we can add more types of maintainers to the `MaintainerChain` later. SPV maintainer, group lifecycles maintainer etc.

Reply:
I added the `MaintainerChain` to the `tbtc.Chain` interface and therefore forced the implementation of `TbtcChain` to have the `BitcoinDifficultyMaintainer()` function.

Then I realise it did not make much sense: we would create the `BitcoinDifficultyMaintainer` and pass it inside the `TbtcChain`, just to return it when we need to use it.
`TbtcChain` would sometimes store that `BitcoinDifficultyMaintainer` (when running `keep-client` as a maintainer) and sometimes it wouldn't (when we use `keep-client` for running keep nodes).

Therefore I treated the `BitcoinDifficultyMaintainer` a separate chain, just like we treat the `TbtcChain` or `BeaconChain`. They all are just handles to the smart contracts.

I put the interface in the `pkg/bitcoin_difficulty/chain.go`, so that it's symmetrical to location of Tbtc interface (`pkg/tbtc/chain.go`).

I put the implementation in the `pkg/chain/ethereum/bitcoin_difficulty`, so that it's symmetrical to the location of Tbtc implementation (`pkg/ethereum/tbtc.go`).

Also I named it `bitcoin_difficulty` for short.